web/src/settings_org.js

```diff
@@ -710,7 +710,12 @@ export function sync_realm_settings(property) {
     }
     const $element = $(`#id_realm_${CSS.escape(property)}`);
     if ($element.length) {
-        discard_realm_property_element_changes($element);
+        const $subsection = $element.closest(".settings-subsection-parent");
+        if ($subsection.find(".save-button-controls").hasClass("hide")) {
+            discard_realm_property_element_changes($element);
+        } else {
+            discard_realm_settings_subsection_changes($subsection);
```

Comment:
We do not call the function to discard all the changes in the subsection when save-discard widget is not visible to avoid unnecessarily updating the elements that were not changed.